compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -488,7 +492,27 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     let mut diagnostics_builder = lint.build(
                         "drop order affected for closure because of `capture_disjoint_fields`",
                     );
-                    diagnostics_builder.note(&migrations_text);
+                    let closure_body_span = self.tcx.hir().span(body_id.hir_id);
+                    let (sugg, app) =
+                        match self.tcx.sess.source_map().span_to_snippet(closure_body_span) {
+                            Ok(s) => (
+                                format!("{{ {}; {} }}", migration_string, s),
+                                Applicability::MachineApplicable,
+                            ),
+                            Err(_) => (migration_string.clone(), Applicability::HasPlaceholders),
+                        };
+
+                    let diagnostic_msg = format!(
+                        "`{}` causes {} to be fully captured",
```

Comment:
maybe "add a dummy let to cause `{}` to be fully captured" would be more concise/readable?

Reply:
I like this, will update the PR